parquet/src/file/metadata.rs

```diff
@@ -349,7 +349,13 @@ impl RowGroupMetaData {
 
     /// Method to convert from Thrift.
     pub fn from_thrift(schema_descr: SchemaDescPtr, mut rg: RowGroup) -> Result<RowGroupMetaData> {
-        assert_eq!(schema_descr.num_columns(), rg.columns.len());
+        if schema_descr.num_columns() != rg.columns.len() {
+            return Err(general_err!(
+                "Column length mismatch. Schema has {} columns while Row Group has {}",
```

Comment:
```suggestion
                "Column count mismatch. Schema has {} columns while Row Group has {}",
```

nit: count seems a better descriptor than length (column length made me first think of size of column, instead of number of columns)